From 32ef5d18178cf21eb01ec516528e0c6006ffab73 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:19:28 -0400
Subject: [fs] nfsd: remove dl_fh field from struct nfs4_delegation

Message-id: <1410898849-22126-85-git-send-email-bfields@redhat.com>
Patchwork-id: 95808
O-Subject: [Patch RHEL7 084/165] nfsd: remove dl_fh field from struct nfs4_delegation
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Jeff Layton <jlayton@primarydata.com>

Now that the nfs4_file has a filehandle in it, we no longer need to
keep a per-delegation copy of it. Switch to using the one in the
nfs4_file instead.

Signed-off-by: Jeff Layton <jlayton@primarydata.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 650ecc8f8ff29a7f0990704f09df232b505b200d
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4callback.c b/fs/nfsd/nfs4callback.c
index c393d6c..e981338 100644
--- a/fs/nfsd/nfs4callback.c
+++ b/fs/nfsd/nfs4callback.c
@@ -337,7 +337,7 @@ static void encode_cb_recall4args(struct xdr_stream *xdr,
  p = xdr_reserve_space(xdr, 4);
  *p++ = xdr_zero;   /* truncate */
 
- encode_nfs_fh4(xdr, &dp->dl_fh);
+ encode_nfs_fh4(xdr, &dp->dl_file->fi_fhandle);
 
  hdr->nops++;
 }
diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index ecfddca..b0f83be 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -604,7 +604,6 @@ alloc_init_deleg(struct nfs4_client *clp, struct svc_fh *current_fh)
  INIT_LIST_HEAD(&dp->dl_perclnt);
  INIT_LIST_HEAD(&dp->dl_recall_lru);
  dp->dl_type = NFS4_OPEN_DELEGATE_READ;
- fh_copy_shallow(&dp->dl_fh, &current_fh->fh_handle);
  INIT_WORK(&dp->dl_recall.cb_work, nfsd4_run_cb_recall);
  return dp;
 out_dec:
@@ -3097,7 +3096,7 @@ void nfsd4_prepare_cb_recall(struct nfs4_delegation *dp)
  struct nfs4_client *clp = dp->dl_stid.sc_client;
  struct nfsd_net *nn = net_generic(clp->net, nfsd_net_id);
 
- block_delegations(&dp->dl_fh);
+ block_delegations(&dp->dl_file->fi_fhandle);
 
  /*
   * We can't do this in nfsd_break_deleg_cb because it is
diff --git a/fs/nfsd/state.h b/fs/nfsd/state.h
index 0097d47..3974773 100644
--- a/fs/nfsd/state.h
+++ b/fs/nfsd/state.h
@@ -96,7 +96,6 @@ struct nfs4_delegation {
  u32   dl_type;
  time_t   dl_time;
 /* For recall: */
- struct knfsd_fh  dl_fh;
  int   dl_retries;
  struct nfsd4_callback dl_recall;
 };
-- 
1.7.1